### PR TITLE
Biometrics login

### LINKED ISSUE #1108: Biometric security
### Describe the feature

instead of using a pin code or a password, let us use the anyhow in the phone integrated biometrics like fingerprint. I suggest that those biometric will need to be reverified after each phone unlock and after a certain time period where the app was not used but the phone was unlocked. Maybe make this a choice also.

### PR DESCRIPTION
**Pull request by channelninja**
# Description

Relates to issue: **ZEUS-1108**

This PR implements Biometry logins for iOS (FaceID, TouchID) and Android (only tested fingerprint unlock).

One security caveat: According to https://blog.nviso.eu/2021/04/06/a-closer-look-at-the-security-of-react-native-biometric-libraries/ is the method I used here (simplePrompt) not very secure since it's only a event based implementation instead of a result based. But as I understand it, the more secure result based one needs a backend server that verifies the result. (which we don't have?)

> A well-implemented result-based biometric authentication, on the other hand, will not be bypassable with tools such as Frida. To implement a secure result-based biometric authentication, the application must use hardware-backed biometric APIs.

closes #1108 

This pull request is categorized as a:

- [x] New feature
- [ ] Bug fix
- [ ] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Quality assurance 
- [ ] Other

## Checklist
- [x] I’ve run `yarn run tsc` and made sure my code compiles correctly. --> Lot's of issues from node_modules? Did I set it up wrong?
- [x] I’ve run `yarn run lint` and made sure my code didn’t contain any problematic patterns
- [x] I’ve run `yarn run prettier` and made sure my code is formatted correctly
- [x] I’ve run `yarn run test` and made sure all of the tests pass

## Testing

If you modified or added a utility file, did you add new unit tests?

- [x] No, how do I unit test biometrics?
- [ ] Yes
- [ ] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [x] Android: Pixel 6 (Android 12)
- [x] iOS: iPhone 13 Pro (iOS 16.2)

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [ ] LND (REST)
- [ ] LND (Lightning Node Connect)
- [ ] Core Lightning (c-lightning-REST)
- [ ] Core Lightning (Spark)
- [ ] Eclair
- [x] LndHub

### Locales
- [x] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the Zeus [Transfix page](https://www.transifex.com/ZeusLN/zeus/) and not directly to this repo

I added only the en.json locale. Is that right?

### Third Party Dependencies and Packages

- [x] Contributors will need to run `yarn` after this PR is merged in
- [x] 3rd party dependencies have been modified:
    * verify that `package.json` and `yarn.lock` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
     "querystring-es3": "0.2.1",
     "react": "18.1.0",
     "react-native": "0.70.6",
+    "react-native-biometrics": "^3.0.1",
```
Comment (kaloudis) [inline package.json:70]: we like to lock version numbers. remove the `^` and run yarn again

Comment (channelninja) [reply package.json]: changed

File: views/Settings/Security.tsx [hunk 2]
```diff
@@ -224,6 +277,35 @@ export default class Security extends React.Component<
...
+                            </ListItem.Title>
+                        </ListItem.Content>
+
+                        <Switch
+                            value={isBiometryEnabled}
+                            onValueChange={async () => {
+                                this.setState({
+                                    isBiometryEnabled: !isBiometryEnabled
+                                });
+                            }}
```
Comment (kaloudis) [inline views/Settings/Security.tsx:305]: call `updateSettings({ isBiometryEnabled: !isBiometryEnabled })` here

File: views/Settings/Security.tsx [hunk 3]
```diff
@@ -119,6 +131,46 @@ export default class Security extends React.Component<
...
+            (prevState.isBiometryEnabled === false &&
+                this.state.isBiometryEnabled) ||
+            (prevState.isBiometryEnabled && !this.state.isBiometryEnabled)
+        ) {
+            this.handleIsBiometryEnabledChange(
+                prevState.isBiometryEnabled,
+                !!this.state.isBiometryEnabled
+            );
+        }
+    }
```
Comment (kaloudis) [inline views/Settings/Security.tsx:172]: no need for these two functions. just call `updateSettings` in the Switch `onValueChange`

File: locales/en.json [hunk 4]
```diff
@@ -457,6 +457,15 @@
...
     "views.Settings.SetDuressPassword.duressPasswordExplanation": "Once set, you can enter your duress password on the login screen to delete all of your node configurations.",
     "views.Settings.SetPin.title": "Set / Change PIN",
+    "views.Settings.Security.FaceID.title": "FaceID",
+    "views.Settings.Security.FaceID.prompt": "FaceID",
+    "views.Settings.Security.TouchID.title": "TouchID",
+    "views.Settings.Security.TouchID.prompt": "TouchID",
+    "views.Settings.Security.Biometrics.title": "Biometrics",
+    "views.Settings.Security.Biometrics.prompt": "Unlock",
+    "views.Lockscreen.FaceID.prompt": "Unlock Zeus",
+    "views.Lockscreen.TouchID.prompt": "Unlock Zeus",
```
Comment (kaloudis) [inline locales/en.json:467]: please try to not reuse values. no need for a translator to translate the same string 2 times

**Comment by channelninja** [comment]
Somehow my comments got added multiple times to different conversations 🤷

**Comment by channelninja** [comment]
https://github.com/SelfLender/react-native-biometrics/issues/240
Could someone with a samsung or huawei device check if FaceId works?

File: ios/zeus/Info.plist [hunk 5]
```diff
@@ -2,6 +2,8 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>NSFaceIDUsageDescription</key>
+	<string>Text explains why you need the FaceID</string>
```
Comment (kaloudis) [inline ios/zeus/Info.plist:6]: We need to fill this out with a proper description

**Review by kaloudis — CHANGES_REQUESTED** [review]

File: views/Lockscreen.tsx [hunk 6]
```diff
@@ -74,6 +75,26 @@ export default class Lockscreen extends React.Component<
         const posEnabled: boolean =
             (settings && settings.pos && settings.pos.squareEnabled) || false;
 
+        if (
+            settings &&
+            settings.isBiometryEnabled &&
+            settings.supportedBiometryType !== undefined &&
```
Comment (kaloudis) [inline views/Lockscreen.tsx:81]: I think the `!== undefined` here is redundant

File: views/Lockscreen.tsx [hunk 7]
```diff
@@ -74,6 +75,26 @@ export default class Lockscreen extends React.Component<
...
+            !modifySecurityScreen
+        ) {
+            const isVerified = await verifyBiometry(
+                localeString('views.Lockscreen.Biometrics.prompt')
+            );
+
+            if (isVerified) {
+                SettingsStore.setLoginStatus(true);
+                navigation.navigate('Wallet');
+                return;
```
Comment (kaloudis) [inline views/Lockscreen.tsx:94]: is this `return` here required?

**Comment by Talej** [comment]
A couple of places use: 

`settings.isBiometryEnabled && settings.supportedBiometryType !== undefined`

Could be a bit cleaner to turn that into a utility helper function.

Checks out over here on iPhone 11 iOS 16.2

**Comment by channelninja** [comment]
@Talej changed it.

**Review by kaloudis — APPROVED** [review]
tACK. Great first contribution!

**Pull request merged**